docs/source/usage.rst

```diff
@@ -0,0 +1,192 @@
+            pydoctor \
+                --add-package=(packagedirectory) \
+                --project-name=(projectname) \
+                --project-url=https://github.com/$GITHUB_REPOSITORY \
+                --html-viewsource-base=https://github.com/$GITHUB_REPOSITORY/tree/$GITHUB_SHA \
+                --make-html \
+                --html-output=./apidocs \
+                --project-base-dir="$(pwd)" \
+                --docformat=restructuredtext 
+
+        - name: Publish pydoctor documentation to the gh-pages branch
+          uses: peaceiris/actions-gh-pages@v3
+          with:
+            github_token: ${{ secrets.GITHUB_TOKEN }}
+            publish_dir: ./apidocs
+            commit_message: "Generate pydoctor documentation"
+
+.. note:: As mentionned in the ``actions-gh-pages`` `documentation`__, the first workflow run won't actually publish the documentation to Github pages. 
+    Github pages needs to be enabled afterwards in the repo settings, select ``gh-pages`` branch, then re-run your workflow. 
+
+    The website we'll be at https://(user).github.io/(repo)/
+
+    __ https://github.com/peaceiris/actions-gh-pages
+
+.. With Sphinx and Read The Docs
+.. ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. .. note:: Documentation to come!
+
+Document part of your package
+-----------------------------
+
+Sometimes, only a couple classes or modules are part of your public API, not all classes and modules need to be documented.  
+
+You can choose to document only a couple classes or modules with the following cumulative configuration option::
+
+  --html-subject=pydoctor.zopeinterface.ZopeInterfaceSystem
+
+This will generate only ``pydoctor.zopeinterface.ZopeInterfaceSystem.html`` and ``objects.inv`` (and CSS and JS files of course). 
+The ``--add-package`` argument still needs to be passed, ``--html-subject`` will act like a filter.  
+
+.. warning:: The ``index.html`` and other index files won't be generated, you need to link to the specific HTML page. 
+
+Sphinx Integration
+------------------
+
+Sphinx object inventories can be used to create links in both ways between
+documentation generated by pydoctor and by Sphinx.
+
+Linking to external API docs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It can link to external API documentation using a Sphinx objects inventory
+with the following cumulative configuration option::
+
+    --intersphinx=https://docs.python.org/3/objects.inv
+
+Linking to your API docs
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+pydoctor's HTML generator will also generate a Sphinx objects inventory
+using the following mapping:
+
+* packages, modules -> py:mod:
+* classes -> py:class:
+* functions -> py:func:
+* methods -> py:meth:
+* attributes -> py:attr:
+
+To use this mapping in Sphinx, configure the `intersphinx extension`__::
+
+    intersphinx_mapping = {
+        'pydoctor': ('http://domain.tld/api', None),
+    }
+
+__ https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html
+
+And use external references::
```

Comment:
I know that this is not yet possible...but just a comment.

In parallel with the block literal it would be nice to also use a real reference so that people can see exactly how it looks.

Maybe something like this

````
Link to a :py:func:`pydoctor function with custom text <pydoctor:pydoctor.model.Documentable.reparent>` 
is used as::

    :py:func:`pydoctor function with custom text <pydoctor:pydoctor.model.Documentable.reparent>

Link to the module with default label :py:mod:`pydoctor:pydoctor.model` ::

    :py:mod:`pydoctor:pydoctor.model`
```

Reply:
I've used twisted's intersphinx for the demo